### PR TITLE
Balder/add control over concurrency and allocation per documentdb.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR

### CODE REVIEW

**Review by geirst — CHANGES_REQUESTED** [review]
Overall looks good. I think we should refactor DocumentDatabaseTestCase.java to use com.yahoo.vespa.model.content.utils.DocType to represent document types.

File: config-model/src/test/java/com/yahoo/vespa/model/search/test/DocumentDatabaseTestCase.java [hunk 1]
```diff
@@ -55,17 +65,26 @@ private String createVespaServices(List<String> sdNames, String mode) {
                       "<content version='1.0' id='test'>\n" +
                       "   <redundancy>1</redundancy>\n");
         retval.append("   <documents>\n");
-        for (String sdName : sdNames) {
-            retval.append("").append("      <document type='").append(sdName).append("' mode='").append(mode).append("'");
+        for (Pair<String, String> nameAndMode : nameAndModes) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config-model/src/test/java/com/yahoo/vespa/model/search/test/DocumentDatabaseTestCase.java]: Fixed

File: config-model/src/test/java/com/yahoo/vespa/model/search/NodeFlavorTuningTest.java [hunk 2]
```diff
@@ -27,6 +32,29 @@ public void require_that_hwinfo_memory_size_is_set() {
...
+        for (Pair<String, String> sdMode : sdAndMode) {
+            builder.documentdb.add(new ProtonConfig.Documentdb.Builder()
+                                   .inputdoctypename(sdMode.getFirst())
+                                   .configid("some/config/id/" + sdMode.getFirst())
+                                   .mode(ProtonConfig.Documentdb.Mode.Enum.valueOf(sdMode.getSecond())));
+        }
+        return configFromMemorySetting(gb, builder);
+    }
+    @Test
+    public void require_that_initial_is_dependent_of_mode() {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config-model/src/test/java/com/yahoo/vespa/model/search/NodeFlavorTuningTest.java]: Fixed

File: config-model/src/main/java/com/yahoo/vespa/model/search/Tuning.java [hunk 3]
```diff
@@ -338,7 +341,7 @@ public void getConfig(ProtonConfig.Builder builder) {
             @Override
             public void getConfig(ProtonConfig.Builder builder) {
                 if (concurrency != null) {
-                    builder.feeding.concurrency(concurrency);
+                    builder.feeding.concurrency(concurrency/2);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/vespa/model/search/Tuning.java]: Fixed

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**